qiskit/pulse/instruction_schedule_map.py

```diff
@@ -155,28 +156,41 @@ def get(self,
             The Schedule defined for the input.
         """
         self.assert_has(instruction, qubits)
-        schedule = self._map[instruction].get(_to_tuple(qubits))
-        if isinstance(schedule, ParameterizedSchedule):
-            schedule = schedule.bind_parameters(*params, **kwparams)
-        return schedule
+        schedule_generator = self._map[instruction].get(_to_tuple(qubits))
 
-    def get_parameters(self, instruction: str, qubits: Union[int, Iterable[int]]) -> Tuple[str]:
-        """Return the list of parameters taken by the given instruction on the given qubits.
+        if callable(schedule_generator):
+            return schedule_generator(*params, **kwparams)
+        # otherwise this is just a Schedule
+        return schedule_generator
+
+    def get_unevaluated(self,
```

Comment:
Can we update this to only return callables? Then the return types are stricter, and the programmer knows what to expect from each method: `get` always returns a `Schedule` and this method always returns a `Callable`. Easy to do by checking what is in `_map`. Then we can also make the name more specific: `get_rule` or `get_callable` or `get_builder`

Reply:
The problem is that Schedule is not callable, I would have to wrap it in a lambda or something.

Reply:
My suggestion is if it is a raw schedule, to error. `get_unevaluated` doesn't make as much sense when the value is already evaluated and cannot be called. Imagine using this as a user: `calibrations.get_unevaluated(...)(*args, **kwargs)` would be pretty natural

Reply:
I imagine this as just a simple accessor method for instrcution+qubits. This is still possible with
```
instr = calibrations.get_unevaluated(...)
if callable(instr):
     instr = instr(*args, **kwargs)
```
The alternative with what you propose is 
```
try:
     instr = calibrations.get_unevaluated(...)(*args, **kwargs)
except:
    instr = calibrations.get(...)
```
please let me know what you think is better and I will implement that.